web/src/components/storage/VolumeDialog.jsx

```diff
@@ -0,0 +1,743 @@
+  }
+
+  /**
+   * @method
+   * @returns {boolean}
+   */
+  check() {
+    const regex = /^swap$|^\/$|^(\/[^/\s]+([^/]*[^/\s])*)+$/;
+    return !regex.test(this.mountPath);
+  }
+
+  /**
+   * @method
+   * @returns {string}
+   */
+  render() {
+    return _("The mount point is invalid");
+  }
+}
+
+class MissingSizeError {
+  /**
+   * @constructor
+   * @param {SizeMethod} sizeMethod
+   * @param {string|number} size
+   */
+  constructor (sizeMethod, size) {
+    this.sizeMethod = sizeMethod;
+    this.size = size;
+  }
+
+  /**
+   * @method
+   * @returns {boolean}
+   */
+  check() {
+    return this.sizeMethod === SIZE_METHODS.MANUAL && !this.size;
+  }
+
+  /**
+   * @method
+   * @returns {string}
+   */
+  render() {
+    return _("A size value is required");
+  }
+}
+
+class MissingMinSizeError {
+  /**
+   * @constructor
+   * @param {SizeMethod} sizeMethod
+   * @param {string|number} minSize
+   */
+  constructor (sizeMethod, minSize) {
+    this.sizeMethod = sizeMethod;
+    this.minSize = minSize;
+  }
+
+  /**
+   * @method
+   * @returns {boolean}
+   */
+  check() {
+    return this.sizeMethod === SIZE_METHODS.RANGE && !this.minSize;
+  }
+
+  /**
+   * @method
+   * @returns {string}
+   */
+  render() {
+    return _("Minimum size is required");
+  }
+}
+
+class InvalidMaxSizeError {
+  /**
+   * @constructor
+   * @param {SizeMethod} sizeMethod
+   * @param {string|number} minSize
+   * @param {string|number} maxSize
+   */
+  constructor (sizeMethod, minSize, maxSize) {
+    this.sizeMethod = sizeMethod;
+    this.minSize = minSize;
+    this.maxSize = maxSize;
+  }
+
+  /**
+   * @method
+   * @returns {boolean}
+   */
+  check() {
+    return this.sizeMethod === SIZE_METHODS.RANGE &&
+      this.maxSize !== -1 &&
+      this.maxSize <= this.minSize;
+  }
+
+  /**
+   * @method
+   * @returns {string}
+   */
+  render() {
+    return _("Maximum must be greater than minimum");
+  }
+}
+
+class ExistingVolumeError {
+  /**
+   * @constructor
+   * @param {string} mountPath
+   * @param {Volume[]} volumes
+   */
+  constructor(mountPath, volumes) {
+    this.mountPath = mountPath;
+    this.volumes = volumes;
+  }
+
+  /**
+   * @method
+   * @returns {Volume|undefined}
+   */
+  findVolume() {
+    return this.volumes.find(t => t.mountPath === this.mountPath);
+  }
+
+  /**
+   * @method
+   * @returns {boolean}
+   */
+  check() {
+    return this.mountPath.length && this.findVolume() !== undefined;
+  }
+
+  /**
+   * @method
+   * @param {(volume: Volume) => void} onClick
+   * @returns {React.ReactElement}
+   */
+  render(onClick) {
+    const volume = this.findVolume();
+    const path = this.mountPath === "/" ? "root" : this.mountPath;
+
+    return (
+      <div className="split">
+        <span>{sprintf(_("There is already a file system for %s."), path)}</span>
+        <Button variant="link" isInline onClick={() => onClick(volume)}>
+          {_("Do you want to edit it?")}
+        </Button>
+      </div>
```

Comment:
Just a note: it LGTM for this iteration, but I plan to make "assisted errors" to look a bit different than "format errors" in a future. The former should have a less "red" (Not be part of FormValidationError) and more prominent look and feel. I would came with a proposal when time permits.